src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -139,10 +137,35 @@ export interface TradeProps {
 
 interface ExpandProps {
   expanded: boolean
-}
+  warning?: 'warning' | 'error'
+}
+
+const ExpanderRow = styled(Row)<{ $expanded: boolean; warning?: 'warning' | 'error' }>`
+  ${({ warning, $expanded }) =>
+    warning &&
+    css`
+      background-color: ${({ theme }) =>
+        $expanded ? 'transparent' : warning === 'error' ? theme.criticalSoft : theme.warningSoft};
+      border-radius: ${({ theme }) => theme.borderRadius.xsmall}rem;
+      padding: 0.375rem 0.5rem 0.375rem 0.375rem;
+      transition: background-color ${AnimationSpeed.Medium} linear, padding ${AnimationSpeed.Medium} linear,
+        width ${AnimationSpeed.Medium} linear;
+    `}
+`
 
-function Expander({ expanded }: ExpandProps) {
-  return <ExpandIcon $expanded={expanded} />
+function Expander({ expanded, warning }: ExpandProps) {
+  return (
+    <ExpanderRow $expanded={expanded} warning={warning} gap={0.5}>
+      {warning && !expanded && (
+        <Tooltip icon={AlertTriangle} iconProps={{ color: warning }} placement="auto">
+          <ThemedText.Caption>
```

Comment:
Is this needed, or is it already supplied by `Tooltip`?

Reply:
this is needed